lib/wallet/wallet.js

```diff
@@ -811,6 +844,23 @@ class Wallet extends EventEmitter {
     }
   }
 
+  /**
+   * Derive an address (doesn't increment depth).
+   * @param {(Number|String)?} acct
+   * @param {Number} branch
+   * @param {Number} index
+   * @returns {Promise} - Returns {@link WalletKey}.
+   */
+
+  async deriveKey(acct, branch, index) {
+    const unlock = await this.writeLock.lock();
```

Comment:
Because we are not writing anything to database there is no need to have `writeLock` here.
You can rename `_deriveKey` to `deriveKey`.